flake.nix

```diff
@@ -22,6 +23,10 @@
       # System types to support.
       supportedSystems = [ "x86_64-linux" ];
 
+      # Nvidia drivers to support
+      supportedNvidiaDrivers = [ "470.103.01" "470.161.03" ];
+      defaultNvidiaDriver = builtins.elemAt supportedNvidiaDrivers 0;        
```

Comment:
we support two drivers and randomly choose the first?  I feel this deserves some explanation

Reply:
those are the two drivers I need at the moment, would you prefer I hard code the preferred version in the default?

Reply:
or like `supportedNvidiaDrivers = defaultNvidiaDriver ++ [ "..." ];

Reply:
maybe add a third variable - defaultNvidiaDriver, extraNvidiaDrivers, and merge them for supportedNvidiaDrivers? then I can remove the redundancy in the filter you noted below. I'm a bit ambivalent though

Reply:
I was going to suggest exactly that (but in my head it was `additionalNvidiaDrivers`).  Up to you though.

Reply:
ok, that sounds good to me

Reply:
opted to write a runtime dispatching wrapper instead